packages/@aws-cdk/aws-dynamodb/test/dynamodb.test.ts

```diff
@@ -719,6 +722,71 @@ test('if an encryption key is included, encrypt/decrypt permissions are added to
   });
 });
 
+test('when specifying STANDARD_INFREQUENT_ACCESS table class', () => {
+  const stack = new Stack();
+  new Table(stack, CONSTRUCT_NAME, {
+    tableName: TABLE_NAME,
+    partitionKey: TABLE_PARTITION_KEY,
+    tableClass: TableClass.STANDARD_INFREQUENT_ACCESS,
+  });
+
+  Template.fromStack(stack).hasResourceProperties('AWS::DynamoDB::Table',
+    {
+      KeySchema: [
+        { AttributeName: 'hashKey', KeyType: 'HASH' },
+      ],
+      TableClass: 'STANDARD_INFREQUENT_ACCESS',
+      AttributeDefinitions: [
+        { AttributeName: 'hashKey', AttributeType: 'S' },
+      ],
+      TableName: 'MyTable',
+    },
+  );
```

Comment:
Weird formatting, plus I don't think there's any reason to assert any properties beyond `TableClass`:

```suggestion
  Template.fromStack(stack).hasResourceProperties('AWS::DynamoDB::Table', {
    TableClass: 'STANDARD_INFREQUENT_ACCESS',
  });
```